### PR TITLE
[sitecore-jss-react] Fix rendering issue in components using withPlaceholder

### PR DESCRIPTION
**Pull request by sc-addypathania**
This fixes the rendering issueWhen additional params are configured in sitecore, the the components using the withPlaceholder HOC crashes because the current implementation of PlaceholderCommon assumes that the params would always be present on any rendering which is not the case as it is an optional property.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/sitecore-jss-react/src/enhancers/withPlaceholder.test.tsx [hunk 1]
```diff
@@ -193,4 +200,29 @@ describe('withPlaceholder HOC', () => {
...
+      it('should render a placeholder with given key when params are not passed', () => {
+        const component = (dataSet.data.sitecore.route.placeholders.main as (
+          | ComponentRendering
+          | RouteData
+        )[]).find((c) => (c as ComponentRendering).componentName);
+        const phKey = 'page-content';
+        const props: PlaceholderProps = {
+          name: phKey,
+          rendering: component,
+          params: { test: 'test' },
```
Comment (art-alexeyenko) [inline packages/sitecore-jss-react/src/enhancers/withPlaceholder.test.tsx:215]: params are passed here for this test. Is this an omission?

**Review by art-alexeyenko — COMMENTED** [review]
Looks like a good solution!
Just one small comment below.

File: packages/sitecore-jss-react/src/enhancers/withPlaceholder.test.tsx (hunk 1 above)
Comment (sc-addypathania) [reply packages/sitecore-jss-react/src/enhancers/withPlaceholder.test.tsx]: Ohh, good catch.
This got left out when I was doing initial testing. 
Removed :)

**Review by art-alexeyenko — APPROVED** [review]
👍

**Pull request merged**